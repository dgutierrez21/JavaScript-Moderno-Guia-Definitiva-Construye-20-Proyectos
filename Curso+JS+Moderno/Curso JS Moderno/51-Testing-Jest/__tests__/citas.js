import Citas from "../js/classes/Citas.js";

describe("Probar la clase de citas", () => {
  const citas = new Citas();

  const id = Date.now();

  test("Agregar nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Firu",
      propietario: "Palmer Pamila",
      telefono: "002211",
      fecha: "5-8-2022",
      hora: "8:00",
      sintomas: "No quiere comer",
    };

    citas.agregarCita(citaObj);

    // Prueba

    expect(citas).toMatchSnapshot();
  });

  test("Actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Rocco",
      propietario: "Palmer Pamila",
      telefono: "002211",
      fecha: "5-8-2022",
      hora: "8:00",
      sintomas: "No quiere comer",
    };

    citas.editarCita(citaActualizada);

    // Prueba
    expect(citas).toMatchSnapshot();
  });

  test("Eliminar cita", () => {
    citas.eliminarCita(id);

    // Prueba
    expect(citas).toMatchSnapshot();
  });
});
