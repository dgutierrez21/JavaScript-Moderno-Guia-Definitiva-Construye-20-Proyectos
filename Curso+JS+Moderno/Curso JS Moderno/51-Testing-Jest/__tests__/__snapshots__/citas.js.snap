// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "5-8-2022",
      "hora": "8:00",
      "id": 1671853519149,
      "mascota": "Rocco",
      "propietario": "Palmer Pamila",
      "sintomas": "No quiere comer",
      "telefono": "002211",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "5-8-2022",
      "hora": "8:00",
      "id": 1671853519149,
      "mascota": "Firu",
      "propietario": "Palmer Pamila",
      "sintomas": "No quiere comer",
      "telefono": "002211",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
